src/Lecture2.hs

```diff
@@ -58,7 +62,8 @@ lazyProduct = error "TODO"
 "ccaabb"
 -}
 duplicate :: [a] -> [a]
-duplicate = error "TODO"
+duplicate []       = []
+duplicate (x : xs) = [x,x] ++ duplicate xs
```

Comment:
It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible 🐌